mithril-aggregator/src/runtime/runner.rs

```diff
@@ -380,64 +293,46 @@ impl AggregatorRunnerTrait for AggregatorRunner {
+            epoch_service
+                .next_aggregate_verification_key()?
                 .to_json_hex()
                 .with_context(|| "convert next avk to json hex failure")?,
         );
 
         Ok(protocol_message)
     }
 
-    async fn create_new_pending_certificate_from_multisigner(
+    async fn create_new_pending_certificate(
         &self,
         beacon: Beacon,
         signed_entity_type: &SignedEntityType,
     ) -> StdResult<CertificatePending> {
-        debug!("RUNNER: create new pending certificate from multisigner");
-        let multi_signer = self.dependencies.multi_signer.read().await;
+        debug!("RUNNER: create new pending certificate");
+        let epoch_service = self.dependencies.epoch_service.read().await;
 
-        let signers = match multi_signer.get_signers().await {
-            Ok(signers) => signers,
-            Err(ProtocolError::Beacon(_)) => vec![],
-            Err(e) => return Err(e.into()),
-        };
-        let next_signers = match multi_signer.get_next_signers_with_stake().await {
-            Ok(signers) => signers,
-            Err(ProtocolError::Beacon(_)) => vec![],
-            Err(e) => return Err(e.into()),
-        };
+        let signers = epoch_service.current_signers_with_stake()?;
+        let next_signers = epoch_service.next_signers_with_stake()?;
 
         let protocol_parameters =
-            multi_signer
-                .get_protocol_parameters()
-                .await?
-                .ok_or_else(|| {
-                    RunnerError::MissingProtocolParameters(format!(
-                        "no current protocol parameters found for beacon {beacon:?}"
-                    ))
+            epoch_service
+                .current_protocol_parameters()
+                .with_context(|| {
+                    format!("no current protocol parameters found for beacon {beacon:?}")
                 })?;
-
-        let next_protocol_parameters = multi_signer
-            .get_next_protocol_parameters()
-            .await?
-            .ok_or_else(|| {
-                RunnerError::MissingProtocolParameters(format!(
-                    "no next protocol parameters found for beacon {beacon:?}"
-                ))
-            })?;
+        let next_protocol_parameters = epoch_service
+            .next_protocol_parameters()
+            .with_context(|| format!("no next protocol parameters found for beacon {beacon:?}"))?;
 
         let pending_certificate = CertificatePending::new(
             beacon,
             signed_entity_type.to_owned(),
-            protocol_parameters.into(),
-            next_protocol_parameters.into(),
-            signers,
-            next_signers.into_iter().map(|s| s.into()).collect(),
+            protocol_parameters.clone(),
+            next_protocol_parameters.clone(),
+            signers.clone().into_iter().map(|s| s.into()).collect(),
+            next_signers.clone().into_iter().map(|s| s.into()).collect(),
```

Comment:
Maybe we could implement (later) conversion functions for `Vec<Signer>` from/to `Vec<SignerWithStakes>`?

Reply:
Since we can't do a `impl From<Vec<Signer>> for Vec<SignerWithStake>` (because the Vec type is not defined by us) I added a `from_vec` method to the Signer entity that can convert any vec of types that implement `Into<Signer>` to a vec of signers.
Is that good for you ?